### PR TITLE
Moved synchronization objects from Gles3Fence to OpenGLFence.

### PR DESCRIPTION
**Pull request by emilwestergren**
Moved Mutex and WaitCondition up an abstraction level. Probably shouldn't be here in the long term but it's nice to be able to begin waiting on a gpu fence even though it has not been created yet on the GPU.

@sebastianbaginski peer

### CODE REVIEW

File: source/draw/gpu/opengl/OpenGLFence.ooc [hunk 1]
```diff
@@ -17,19 +17,37 @@
...
+import threading/[Mutex, WaitCondition]
 
 version(!gpuOff) {
 OpenGLFence: class extends GpuFence {
 	_backend: GLFence
-	init: func (context: OpenGLContext) {
-		this _backend = context backend as GLContext createFence()
-	}
+	_syncCondition := WaitCondition new()
+	_mutex := Mutex new()
```
Comment (sebastianbaginski) [inline source/draw/gpu/opengl/OpenGLFence.ooc]: We should maybe create a class for these two (`WaitCondition` + `Mutex`), it is a common pair to use. Any suggestions for name ?

Comment (emilwestergren) [inline source/draw/gpu/opengl/OpenGLFence.ooc]: Sounds like a good idea, I think @simonmika agrees. Regarding name I'm not sure.
* `SynchronizationLock`
* `WaitLock`
* `BlockLock` (Catchy right?)

Do we want to keep `signal` `wait` and `broadcast` or create a new interface?

Comment (sebastianbaginski) [inline source/draw/gpu/opengl/OpenGLFence.ooc]: I vote for keeping the interface, as everyone is already familiar with it.
`MutexCondition`, `WaitLock`, `SynchronizedLock`
yes, I guess nothing beats `BlockLock`

Comment (marcusnaslund) [inline source/draw/gpu/opengl/OpenGLFence.ooc]: I like `WaitLock` but it may be too close to `WaitCondition`. `SynchronizationLock` is also good.

Comment (emilwestergren) [inline source/draw/gpu/opengl/OpenGLFence.ooc]: https://github.com/cogneco/ooc-kean/issues/1043

**Pull request merged**